src/rpdk/core/data_loaders.py

```diff
@@ -107,12 +139,25 @@ def load_resource_spec(resource_spec_file):  # noqa: C901
         raise SpecValidationError(str(e)) from e
 
     validator = make_resource_validator()
+    additional_properties_validator = (
+        make_resource_validator_with_additional_properties_check()
+    )
     try:
         validator.validate(resource_spec)
     except ValidationError as e:
         LOG.debug("Resource spec validation failed", exc_info=True)
         raise SpecValidationError(str(e)) from e
 
+    try:
+        additional_properties_validator.validate(resource_spec)
+    except ValidationError as e:
+        LOG.warning(
+            "[Warning] Resource spec validation would fail from next \
+major version. Provider should mark additionalProperties as false if the \
+property is of object type and has properties or patternProperties defined \
+in it. Please fix the warnings: %s",
+            str(e),
```

Comment:
This generates thousands of lines of output [when run across all existing resource schemas](https://github.com/aws-cloudformation/cloudformation-cli/pull/604#issuecomment-756521889). Wondering if just outputting the `message` (`'additionalProperties' is a required property`) instead would be less noisy:
```suggestion
            e.message,
```